spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/SparkTemplate.scala

```diff
@@ -4,7 +4,16 @@ import org.apache.spark.{SparkEnv, SparkConf, SparkContext}
 
 trait SparkTemplate {
   val conf = SparkTemplate.conf
-  val sc = SparkTemplate.sc
+  def sc = SparkTemplate.sc
+
+  def resetSparkContext(conf: SparkConf = conf): Unit = {
```

Comment:
So just import companion object. Methods modifying vars should be close to vars. If var is declared on the companion object, all methods manipulating that state should be on the same object.